proxy/src/main/java/net/md_5/bungee/util/PacketLimiter.java

```diff
@@ -0,0 +1,43 @@
+package net.md_5.bungee.util;
+
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+
+@RequiredArgsConstructor
+public class PacketLimiter
+{
+    // max amount of packets allowed per second
+    private final int limit;
+    // max amount of data allowed per second
+    private final int dataLimit;
+
+    @Getter
+    private int counter;
+    @Getter
+    private int dataCounter;
+    private long nextSecond;
+
+    /**
+     * Counts the received packet amount and size
+     * @param size size of the packet
+     * @return return false if the player should be kicked
+     */
+    public boolean received(int size)
```

Comment:
i'm not happy with the method name, but i also have no ideas of a better name, ifeel with this name, the return value is not intuitive

Reply:
Any other names that comes to my mind sounds weird and as it is only called by the handler boss and is not api i didn‘t really care

but if you have a better name we can change it, but it doesnt bother me if it stays like this

Reply:
maybe i could rename to "check" or "incrementAndCheck"